Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/NetworkSearchController.java

```diff
@@ -96,4 +110,51 @@ public String getSearch(@PathVariable("projectUnixName") String projectUnixName,
         return new ResponseEntity<String>(defaultJsonErrorMsg, HttpStatus.INTERNAL_SERVER_ERROR);
     }
 
+    @RequestMapping(value = "sites/{projectUnixName}/networks/search", method = RequestMethod.GET)
+    public String getSearchTransformedNetwork(@PathVariable("projectUnixName") String projectUnixName,
+                                              @RequestParam("conceptId") String conceptId,
+                                              Model model)
+        throws QuadrigaStorageException {
+
+        IProject project = projectManager.getProjectDetailsByUnixName(projectUnixName);
+        if (project == null) {
+            return "forbidden";
+        }
+
+        ITransformedNetwork transformedNetwork = transformationManager.getSearchTransformedNetwork(
+                project.getProjectId(), conceptId);
+
+        String json = null;
+        if (transformedNetwork != null) {
+            json = d3Creator.getD3JSON(transformedNetwork.getNodes(), transformedNetwork.getLinks());
+        }
+
+        String searchNodeLabel = "";
+        if (transformedNetwork == null || transformedNetwork.getNodes().size() == 0) {
+            model.addAttribute("isNetworkEmpty", true);
+        } else {
+            // get the search concept id
+            for (Node node: transformedNetwork.getNodes().values()) {
+                if (conceptId.equals(node.getConceptId())) {
+                    searchNodeLabel = node.getLabel();
```

Comment:
Why don't you get the label from the conceptpower reply (when you get the description)? It's unnecessary to iterate over all the nodes again.